src/app/auth/permission.service.ts

```diff
@@ -0,0 +1,63 @@
+import { Injectable } from '@angular/core';
+import { JwtHelperService } from '@auth0/angular-jwt';
+
+export type Permission = {
+  exp: number,
+  resource_set_id: string,
+  resource_set_name: string,
+  scopes: Array<string>
+};
+
+@Injectable()
+export class PermissionService {
+  jwtHelper: JwtHelperService;
+  constructor() {
+    this.jwtHelper = new JwtHelperService();
+  }
+
+  /**
+   * Returns all the scopes a user has for a specific resource.
+   * @param resourceId ID of a specific resource such as a Space
+   */
+  getAllScopes(resourceId: string): Array<string> {
+    const permissions = this.getPermission(resourceId);
+    return permissions ? permissions.scopes : [];
+  }
+
+  /**
+   * Checks if a user has a specific scope for a resource.
+   * @param resourceId ID of a specific resource such as a Space
+   * @param scope the scope you want to check for. Ex - `can edit`
+   */
+  checkScope(resourceId: string, scope: string): boolean {
+    const permissions = this.getPermission(resourceId);
+    return permissions ? permissions.scopes.includes(scope) : false;
+  }
+
+  /**
+   * Returns the permission for a specific resource.
+   * @param resourceId ID of a specific resource such as a Space
+   */
+  getPermission(resourceId: string): Permission {
+    const decodedToken = this.getDecodedToken();
+    if (this.isValidRPT(decodedToken)) {
+      return decodedToken.permissions.find((permission: Permission) => permission.resource_set_id === resourceId);
+    }
+  }
+
+  /**
+   * Decodes the JWT token using JwtHelperService from `angular-jwt`.
+   */
+  getDecodedToken() {
```

Comment:
Make private? I see it's used in a test but does it make sense for this to be API?